docs/crd.md

```diff
@@ -0,0 +1,49 @@
+# Restoring Deleted CRs
+
+When a Rook CR is deleted, the Rook operator will respond to the deletion event to attempt to clean up the cluster resources. If any data is still present in the cluster, Rook will refuse to delete the CR to ensure data is not lost. The operator will refuse to remove the finalizer on the CR until the underlying data is deleted.
+
+While the underlying Ceph data and daemons continue to be available, the CRs will be stuck indefinitely in a Deleting state in which the operator will not continue to ensure cluster health. Upgrades will be blocked, further updates to the CRs are prevented, and so on. Since Kubernetes does not allow undeleting resources, the command below will allow repairing the CRs without even necessarily suffering cluster downtime.
+
+## Restore Command
+
+```bash
+kubectl rook-ceph restore-deleted <CRD> [CRName]
+
+# `restore-deleted`: is the command to restore the deleted CR
+# `<CRD>`: the CRD type that is to be restored, such as CephCluster, CephFilesystem, CephBlockPool and so on.
+# `[CRName]`: is the name of the specific CR which you want to restore since there can be multiple instances under the same CRD. For example, if there are multiple CephFilesystems stuck in deleting state, a specific filesystem can be restored: `restore-deleted cephfilesystem filesystem-2`.
+```
+
+## CephCluster Restore Example
+
+```bash
+kubectl rook-ceph restore-deleted cephcluster [cephClusterName]
+
+Info: Detecting which ceph crd resources to restore
+Info: Restoring crd cephcluster name my-cluster
```

Comment:
After we detect which resource(s) to restore, I'm wondering if we should pause and confirm. There is some risk to this operation, so it felt very sudden to continue without any prompt. For example, prompt:
 "The resource %s was found deleted. Do you want to restore it? yes | no". Only if they type `y` or `yes` then we would continue. It's not super risky so we don't need a longer confirmation string, but something simple like this seems best. Thoughts?

Reply:
that's a good point and I agree this is a kind of destructive command better confirmed by user input.

And `yes` is better than just `y` for input.